src/app/complex/complex-details/complex-expression/complex-expression.component.ts

```diff
@@ -8,100 +7,50 @@ import {MIJson} from 'complexviewer';
   templateUrl: './complex-expression.component.html',
   styleUrls: ['./complex-expression.component.scss']
 })
-export class ComplexExpressionComponent implements OnInit {
-  private _gxa = null;
-  private _participants: Participant[];
-  private _complexSpecies: string;
-  private _crossReferences: CrossReference[];
-  private _gxaParamsQueries: string;
-  private _goCellularXRefs: CrossReference[];
+export class ComplexExpressionComponent {
 
+  gxa = input<any>();
+  complexSpecies = input<string>();
+  crossReferences = input<CrossReference[]>();
   complexMIJSON = input<MIJson>();
 
-  constructor() {
-  }
+  gxaParamsQueries: string;
+  goCellularXRefs: CrossReference[];
+  cellLines: CrossReference[];
 
-  ngOnInit() {
-    this.findXRefs();
-    this.findGXAQueries();
+  constructor() {
+    effect(() => this.findXRefs(this.crossReferences()));
```

Comment:
Will duplicate entries within the different fields if input changes, since we just are just pushing to the stable array object. Ideally refactor to use computed( instead of using effects)